### PR TITLE
[HIVE-26291] fix fd leak in ranger rest client

### PR DESCRIPTION
**Pull request by adrian-wang**
### What changes were proposed in this pull request?

fix fd leak when read ranger policy from json file

### Why are the changes needed?

fd should be closed when done using.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

N/A.

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ranger/RangerRestClientImpl.java [hunk 1]
```diff
@@ -428,9 +428,7 @@ public RangerExportPolicyList readRangerPoliciesFromJsonFile(Path filePath,
                                                                HiveConf conf) throws SemanticException {
     RangerExportPolicyList rangerExportPolicyList = null;
     Gson gsonBuilder = new GsonBuilder().setDateFormat("yyyyMMdd-HH:mm:ss.SSS-Z").setPrettyPrinting().create();
-    try {
-      FileSystem fs = filePath.getFileSystem(conf);
-      InputStream inputStream = fs.open(filePath);
+    try (InputStream inputStream = filePath.getFileSystem(conf).open(filePath)) {
       Reader reader = new InputStreamReader(inputStream, Charset.forName("UTF-8"));
```
Comment (slachiewicz) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ranger/RangerRestClientImpl.java:432]: Reader also should be autoclosed

Comment (adrian-wang) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ranger/RangerRestClientImpl.java]: thanks, code updated now

File: ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ranger/RangerRestClientImpl.java [hunk 2]
```diff
@@ -428,10 +428,7 @@ public RangerExportPolicyList readRangerPoliciesFromJsonFile(Path filePath,
                                                                HiveConf conf) throws SemanticException {
     RangerExportPolicyList rangerExportPolicyList = null;
     Gson gsonBuilder = new GsonBuilder().setDateFormat("yyyyMMdd-HH:mm:ss.SSS-Z").setPrettyPrinting().create();
-    try {
-      FileSystem fs = filePath.getFileSystem(conf);
-      InputStream inputStream = fs.open(filePath);
-      Reader reader = new InputStreamReader(inputStream, Charset.forName("UTF-8"));
+    try (Reader reader = new InputStreamReader(filePath.getFileSystem(conf).open(filePath), Charset.forName("UTF-8"))) {
```
Comment (slachiewicz) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ranger/RangerRestClientImpl.java:431]: still leaks InputStream - in try should be line for InputStream and Reader

Comment (adrian-wang) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ranger/RangerRestClientImpl.java]: @slachiewicz Thanks for your review, I've updated the patch

**Review by slachiewicz — APPROVED** [review]

**Pull request closed without merging**